### ISSUE TITLE
Bind to FastCGI Port again after FastCGI Shutdown

### ISSUE THREAD

**Issue by tobya** [mholt/caddy#1166]
(Are you asking for help with using Caddy? Please use our forum instead: https://forum.caddyserver.com. If you are filing a bug report, please answer the following questions. If your issue is not a bug report, you do not need to use this template. Either way, please consider donating if we've helped you. Thanks!)

### 1. What version of Caddy are you running (`caddy -version`)?
0.9.2 on Windows 2012

### 2. What are you trying to do?
Run a php website via fastcgi 

### 3. What is your entire Caddyfile?
```text
 site.example.ie {

 
   log    c:\logs\access.log
   errors c:\logs\errors.log
   

  rewrite / {
   r (.*)
   to /index.php
  }

   fastcgi / 127.0.0.1:9876 php
   root "C:\BCSDEV\site\public"

 }
```

### 4. How did you run Caddy (give the full command and describe the execution environment)?

C:\caddy\caddy.exe -conf="C:\utilities\caddy\caddyfile"
### 5. What did you expect to see?

My website stay up for days and days

### 6. What did you see instead (give full error messages and/or log)?
It goes down after n requests.

php fastcgi exits and when it is restarted, even if the port is the same caddy won't connect to the port.  You get a series of 502 bad gatewary errors until fixed.

I need to shut down caddy, restart fastcgi and then restart caddy.  Is it possible to get caddy to bind to the fastcgi port even if it went away briefly?

By default php fastcgi on windows exits after 500 requests to prevent memory issues.  You can prevent this by 
SET PHP_FCGI_MAX_REQUESTS=0

However this may lead to memory issues.

There are scripts for restarting fastcgi when it closes, but caddy won't be able to use it anymore.  It would be great if caddy could reconnect without needing to restart.

### 7. How can someone who is starting from scratch reproduce this behavior as minimally as possible?

It should be, just set up a php fastcgi site on windows

**Comment by elcore** [comment]
Hello,

please upgrade to Caddy 0.9.3, report back if this still happens or solves your issue.

**Comment by tobya** [comment]
Appears to be fixed in 0.9.3 thanks
